lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -0,0 +1,677 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""In this class, we manage backup configurations and actions.
+
+Specifically backups are handled with Percona Backup MongoDB (pbm).
+A user for PBM is created when MongoDB is first started during the start phase.
+This user is named "backup".
+"""
+
+import json
+import logging
+import re
+import subprocess
+import time
+from typing import Dict, List, Optional
+
+from charms.data_platform_libs.v0.s3 import CredentialsChangedEvent, S3Requirer
+from charms.mongodb.v0.helpers import (
+    current_pbm_op,
+    process_pbm_error,
+    process_pbm_error_k8s,
+    process_pbm_status,
+)
+from charms.operator_libs_linux.v2 import snap
+from ops.framework import Object
+from ops.model import (
+    ActiveStatus,
+    BlockedStatus,
+    MaintenanceStatus,
+    StatusBase,
+    WaitingStatus,
+)
+from ops.pebble import ExecError
+from tenacity import (
+    Retrying,
+    before_log,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_fixed,
+)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "18c461132b824ace91af0d7abe85f40e"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 8
```

Comment:
```
> charmcraft fetch-lib charms.mongodb.v0.mongodb_backups
Library charms.mongodb.v0.mongodb_backups version 0.6 downloaded.
```
Like with the other lib, I think charm hub will complain for skipping a version. I guess we can publish manual versions to catch up if it does.